### ISSUE TITLE
iOS - full screen mode interferes with iOS controls

### ISSUE THREAD

**Issue by klslz** [CDrummond/lms-material#804]
When using iOS Safari in full screen mode, which gives MS an app-like look,  the bottom controls /Playing/Browse/Queue get too close the the edge and tend to interfere with the iOS Windows control ( e.g. white bar (Home Indicator)).  I am wondering if the bottom MS controls can be shifted slightly upwards. 
The string "Playing" is in my case completely covered by the whitebar

I am using Safari and latest (17.2.) iOS.

**Comment by klslz** [comment]
As already mentioned, the whitebar fully overlaps (covers up) the "Playing" string. I'll send a screenshot later.

**Comment by CDrummond** [comment]
D'oh!!!, Sorry missed the word "completely"! No need for a screenshot.

This is most odd, as the code should detect that its fullscreen and add 12px worth of padding to work-around this.

Is this an iPhone or iPad?

**Comment by klslz** [comment]
It's an iPhone 13Pro Max

**Comment by CDrummond** [comment]
Please can you update yopur checkout, restart LMS, and try to add to homescreen again? As per the icon issue I have updated the code so that when the manifest is served to iOS it also adds `?addpad` to the URL. This query parameter forces MaterialSkin to add this 12px padding.

**Comment by klslz** [comment]
Didn't work.

I removed related bookmarks, homescreen icon and also removed cache and all open browser windows.
That's what I do to get rid of everything. All this is also required to get the icons properly updated btw.

**Comment by CDrummond** [comment]
Wait, so this is not adding to homescreen - just asking Safari to go fullscreen?

**Comment by klslz** [comment]
No. If Safari is fullscreen, there is the overlap. If done via icon on homescreen or through the browser.

That add-to-homescreen doesn’t goes into fullscreen automatically is the other issue I reported.

**Comment by CDrummond** [comment]
Can you please update and try again? I've added code to (try to) detect fullscreen changes.

**Comment by klslz** [comment]
No luck. No change.

**Comment by klslz** [comment]
Perhaps there’s some kind of scaling messing around?!?

**Comment by CDrummond** [comment]
I tried on my wife's phone. I assume you have placed the Safari addressbar at the top of the screen, and then use the expand option? If I do that then I can see this, but its not really 'fullscreen' - as there's still the address details. If Safari does not let the page know its fullscreen then not sure what todo.

**Comment by CDrummond** [comment]
Giving up. Updated code to always add this padding, rather than only when required.

**Comment by klslz** [comment]
Hmmh. Perhaps an issue is that somehow the whitebar remains active without anybody touching the screen.
Usually it disappears after a short while. Q is: What keeps it active?

**Comment by CDrummond** [comment]
Sorry, don't care - spending **way** too much time on silly iOS issues.

**Comment by CDrummond** [comment]
Can you try branch `ios-nav` ? Does this hide it when used in browser? What about when added to home screen?

**Comment by klslz** [comment]
Looks good now👍 Even with Safari navigation on top the whitebar sits a couple of pixels below “Playing”.  It’s still permanent on though - no autohide.
And MS also opens in fullscreen now👍
It now also won’t occupy Safari at the same time anymore. It seems to act like a web app now. Great. Anything else I can test?

**Comment by CDrummond** [comment]
...that's the `ios-nav` branch? Odd I tested on my wife's iPhone and white bar still overlapped text when not in app mode.

**Comment by CDrummond** [comment]
I've merged most of this into master, can you confirm this is now OK? If it also works in app mode then please close the other issue.

**Comment by klslz** [comment]
![IMG_2278](https://github.com/CDrummond/lms-material/assets/5501886/667ba814-75bd-46cc-83e5-5ed7202a4028)

Yep. iOS-nav

**Comment by klslz** [comment]
I’ll test the master tomorrow.

**Comment by klslz** [comment]
Just checked with latest master checkout. It still works. Case closed. THX a lot.
